File: Project Euler #15/Java/Solution15.java
import java.math.BigInteger;
import java.util.Scanner;

public class Solution15 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-->0){
            BigInteger mul1 = new BigInteger("1");
            BigInteger mul2 = new BigInteger("1");
            BigInteger mul3 = new BigInteger("1");
            int n = sc.nextInt();
            int m = sc.nextInt();
            for(int i=2;i<=n;i++){
                mul1 = mul1.multiply(new BigInteger(i+""));
            }
            for(int i=2;i<=m;i++){
                mul2 = mul2.multiply(new BigInteger(i+""));
            }
            for(int i=2;i<=n+m;i++){
                mul3 = mul3.multiply(new BigInteger(i+""));
            }
            mul1 = mul1.multiply(mul2);
            mul3 = mul3.divide(mul1);
            mul3 = mul3.mod(new BigInteger(1000000007+""));
            System.out.println(mul3.toString());
        }
        
    }
}

File: Project Euler #15/Python/solution15.py
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        n , m = list(map(int, input().split()))
        mul1 = 1
        mul2 = 1
        mul3 = 1
        for j in range(1, n+1,1):
            mul1 = mul1*j
        for j in range(1, m+1,1):
            mul2 = mul2*j
        for j in range(1, n+m+1, 1):
            mul3 = mul3*j
        print((mul3//(mul2*mul1))%1000000007)           
